restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1727,11 +1789,39 @@ private void outboundToPstn(final CreateCall request, final ActorRef sender) thr
 
 
     private void outboundToSip(final CreateCall request, final ActorRef sender) throws ServletParseException {
+        final String uri = (request.getOutboundProxy()!= null && (!request.getOutboundProxy().isEmpty())) ? request.getOutboundProxy() : "";
         SipURI outboundIntf = null;
         SipURI from = null;
         SipURI to = null;
 
-        to = (SipURI) sipFactory.createURI(request.to());
+        String toUser = "";
+        String toHost = "";
+        String scheme = "";
+        boolean isSecure = false;
+        if(request.to().contains("@") &&
```

Comment:
two reasons i see
- rcml couldve been defined with either 
`<Sip>sip:+1234@host.com:5070</Sip>`
or
`<Sip>sips:+1234@host.com:5070</Sip>`

- and we need to replace (or not) the host part with whatever host we want to replace it with

But I guess i can see what you prefer to do, is it with two calls to `SipFactory`?
```
final String uri = (request.getOutboundProxy()!= null && (!request.getOutboundProxy().isEmpty())) ? request.getOutboundProxy() : "";
SipURI to = (SipURI) sipFactory.createURI(request.to());

if(!uri.isEmpty() {
    String [] tokens = request.to().split("[@:]");
    String toUser = tokens[1];
    String toHost = tokens[2];
    boolean isSecure = to.isSecure();            
    to = sipFactory.createSipURI(toUser, toHost);
    to.setSecure(isSecure);
}
```
is this preferable?